src/SFA.DAS.Provider.PR.Web/Controllers/AddEmployer/SearchByEmailController.cs

```diff
@@ -74,17 +76,37 @@ public async Task<IActionResult> Index([FromRoute] int ukprn, SearchByEmailSubmi
             return RedirectToRoute(RouteNames.AddEmployerMultipleAccounts, new { ukprn });
         }
 
-        if (relationshipByEmail.HasRelationship != null && !relationshipByEmail.HasRelationship.Value)
-        {
-            sessionModel.AccountLegalEntityId = relationshipByEmail.AccountLegalEntityId;
-            sessionModel.AccountLegalEntityName = relationshipByEmail.AccountLegalEntityName;
-            sessionModel.AccountId = relationshipByEmail.AccountId;
-            _sessionService.Set(sessionModel);
+        sessionModel.AccountLegalEntityId = relationshipByEmail.AccountLegalEntityId;
+        sessionModel.AccountLegalEntityName = relationshipByEmail.AccountLegalEntityName;
+        sessionModel.AccountId = relationshipByEmail.AccountId;
+
+        _sessionService.Set(sessionModel);
+
+        return RedirectToRoute(relationshipByEmail.HasRelationship is true
+            ? RouteNames.EmailLinkedToAccountWithRelationship :
+            RouteNames.OneAccountNoRelationship, new { ukprn });
+    }
+
+    [HttpGet]
+    [Route("emailLinkedToAccountWithRelationship", Name = RouteNames.EmailLinkedToAccountWithRelationship)]
+    public IActionResult EmailLinkedToAccountWithRelationshipShutterPage([FromRoute] int ukprn)
+    {
+        var sessionModel = _sessionService.Get<AddEmployerSessionModel>();
+        var email = sessionModel?.Email;
+        var employerName = sessionModel?.AccountLegalEntityName!;
 
-            return RedirectToRoute(RouteNames.OneAccountNoRelationship, new { ukprn });
+        if (string.IsNullOrEmpty(email) || string.IsNullOrEmpty(employerName) || sessionModel!.AccountLegalEntityId == null)
+        {
+            return RedirectToRoute(RouteNames.AddEmployerStart, new { ukprn });
         }
 
-        return RedirectToRoute(RouteNames.AddEmployerSearchByEmail, new { ukprn });
+        var accountLegalEntityId = sessionModel.AccountLegalEntityId.Value;
```

Comment:
I see no benefit of creating new properties to match the session variables (excluding accountLegalEntityIdEncoded), could probably tidy this up by removing the additional properties and accessing the session variable directly.

Reply:
that's fair